src/coreclr/tools/Common/TypeSystem/Ecma/SymbolReader/UnmanagedPdbSymbolReader.cs

```diff
@@ -364,10 +364,782 @@ private void DisposeInternal()
 
         interface ISymUnmanagedReader
         {
-            int GetMethod(int methodToken, out ISymUnmanagedMethod method);
+            int GetMethod(int methodToken, out ISymUnmanagedMethod? method);
             // No more members are used
         }
 
+        private sealed class SymUnmanagedNamespaceWrapperCache : ComWrappers
+        {
+            public static readonly SymUnmanagedNamespaceWrapperCache Instance = new SymUnmanagedNamespaceWrapperCache();
+
+            protected override unsafe ComInterfaceEntry* ComputeVtables(object obj, CreateComInterfaceFlags flags, out int count) => throw new NotImplementedException();
+            protected override object? CreateObject(IntPtr externalComObject, CreateObjectFlags flags)
+            {
+                Debug.Assert(flags == CreateObjectFlags.UniqueInstance);
+                return new SymUnmanagedNamespaceRcw(externalComObject);
+            }
+            protected override void ReleaseObjects(IEnumerable objects) => throw new NotImplementedException();
+
+            public unsafe sealed record SymUnmanagedNamespaceRcw(IntPtr Inst) : ISymUnmanagedNamespace
+            {
+                private bool _disposed = false;
+
+                public int GetName(int bufferLength, out int count, char[] name)
+                {
+                    var func = (delegate* unmanaged<IntPtr, int, int*, char*, int>)(*(*(void***)Inst + 3));
+                    int hr;
+
+                    fixed (int* countPtr = &count)
```

Comment:
Should I pin the locals? Or can I just pass the `out count` as is and use `delegate* unmanaged<IntPtr, int, out int, char*, int` for such cases?